src/cargo/ops/cargo_rustc/fingerprint.rs

```diff
@@ -454,17 +450,19 @@ pub fn prepare_build_cmd<'a, 'cfg>(cx: &mut Context<'a, 'cfg>, unit: &Unit<'a>)
     // necessary for that fingerprint.
     //
     // Hence, if there were some `rerun-if-changed` directives forcibly change
-    // the kind of fingerprint over to the `MtimeBased` variant where the
-    // relevant mtime is the output path of the build script.
+    // the kind of fingerprint by reinterpreting the dependencies output by the
+    // build script.
     let state = cx.build_state.clone();
     let key = (unit.pkg.package_id().clone(), unit.kind);
+    let root = unit.pkg.root().to_path_buf();
     let write_fingerprint = Work::new(move |_| {
         if let Some(output_path) = output_path {
             let outputs = state.outputs.lock().unwrap();
-            if !outputs[&key].rerun_if_changed.is_empty() {
-                let slot = MtimeSlot(Mutex::new(None));
-                fingerprint.local = LocalFingerprint::MtimeBased(slot,
-                                                                 output_path);
```

Comment:
Er actually I had to keep `update_local()?`. I don't know why and don't quite have time to investigate right now. I'm also not sure I understand your comment though, can you elaborate?